site/src/Librecores/ProjectRepoBundle/Doctrine/ProjectMetricsProvider.php

```diff
@@ -238,4 +267,290 @@ function (LanguageStat $a, LanguageStat $b) {
+            $score += 2;
+        } else {
+            $score -= 1;
+        }
+
+        // Max +2.5
+        // Even though it is GitHub specific, it is user feedback and hence
+        // a real indicator of project quality
+        $stars = $project->getStars();
+
+        if ($stars > 10000) {
+            $score += 2.5;
+        } else if ($stars > 1000) {
+            $score += 1;
+        } else if ($stars > 100) {
+            $score += 0.5;
+        }
+
+        // TODO: +0.5 for release tags
+
+        // +0.25 for changelog
+        if (preg_match('/change\s?log|release\s?(notes|history)?/i', $project->getDescriptionText()))
+        {
+            $score += 0.25;
+        }
+
+        // TODO: Handle repositories with CHANGELOG in a file
+
+        // +1 for description
+        // It is important for a project to have a description
+        if ($project->getDescriptionText()) {
+            $score += 1;
+        } else {
+            $score -= 2;
+        }
+
+        // +1 for a license file
+        // -3 since this is very important for open source projects
+        if ($project->getLicenseText()) {
+            $score += 1;
+        } else {
+            $score -= 3;
+        }
+
+        // TODO: This section needs improvement
+
+        // commit activity
+        $commitActivity = $this->getCommitActivityTrend($project);
+        $averageCommits = StatsUtil::normalize($this->getCommitCountAverage($project));
+
+        // +0.5 for stable project with mid term > 0.2
+        if ($averageCommits['mid'] > 0.2) {
+            $score += 0.5;
+        } else {
+            $score -= 0.5;
+        }
+
+        // +0.5 for constant commits
+        if ($averageCommits['end'] > 0.05) {
+            $score += 0.5;
+        } else {
+            $score -= 0.5;
+        }
+
+        // +0.5 if commit activity is not decreasing
+
+        if ($commitActivity['mid'] >= 0) {
+            $score += 0.25;
+        } else {
+            $score -= 0.25;
+        }
+
+        if ($commitActivity['end'] >= 0) {
+            $score += 0.25;
+        } else {
+            $score -= 0.25;
+        }
+
+        // +0.5 if project is still interesting
+
+        if ($this->getContributorActivityTrend($project) > 0.3) {
+            $score += 0.5;
+        }
+
+        return max(ceil($score * 5/17.5), 0);
+    }
+
+
+    /**
+     * Calculate trends in commit activity
+     *
+     * @param Project $project
+     * @return array
+     */
+    public function getCommitActivityTrend(Project $project)
+    {
+        $yearlyCommitCount = array_values($this->commitRepository->
+            getCommitHistogram($project->getSourceRepo(), Dates::INTERVAL_YEAR));
+
+        if (empty($yearlyCommitCount)) {
+
+            return [
+                'start' => -1,
+                'mid' => -1,
+                'end' => -1
+            ];
+        }
+
+        $length = count($yearlyCommitCount);
```

Comment:
Maybe rename `$length` to something more descriptive? something like `$yearsWithCommits`